packages/core/src/hooks/form/index.ts

```diff
@@ -1 +1,341 @@
+  const redirect: UseFormReturnType["redirect"] = (
+    redirect = isEdit ? "list" : "edit",
+    redirectId = id,
+    routeParams = {},
+  ) => {
+    handleSubmitWithRedirect({
+      redirect: redirect,
+      resource,
+      id: redirectId,
+      meta: { ...pickedMeta, ...routeParams },
+    });
+  };
+
+  const queryResult = useOne<TQueryFnData, TError, TData>({
+    resource: identifier,
+    id,
+    queryOptions: {
+      // Only enable the query if it's not a create action and the `id` is defined
+      enabled: !isCreate && id !== undefined,
+      ...props.queryOptions,
+    },
+    liveMode: props.liveMode,
+    onLiveEvent: props.onLiveEvent,
+    liveParams: props.liveParams,
+    meta: { ...combinedMeta, ...props.queryMeta },
+    dataProviderName: props.dataProviderName,
+  });
+
+  const createMutation = useCreate<TResponse, TResponseError, TVariables>({
+    mutationOptions: props.createMutationOptions,
+  });
+
+  const updateMutation = useUpdate<TResponse, TResponseError, TVariables>({
+    mutationOptions: props.updateMutationOptions,
+  });
+
+  const mutationResult = isEdit ? updateMutation : createMutation;
+  const isMutationLoading = mutationResult.isLoading;
+  const formLoading = isMutationLoading || queryResult.isFetching;
+
+  const { elapsedTime } = useLoadingOvertime({
+    isLoading: formLoading,
+    interval: props.overtimeOptions?.interval,
+    onInterval: props.overtimeOptions?.onInterval,
+  });
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: This is a controlled unmounting effect.
+  React.useEffect(() => {
+    // After `autosaved` is set to `true`, it won't be set to `false` again.
+    // Therefore, the `invalidate` function will be called only once at the end of the hooks lifecycle.
+    return () => {
+      if (props.autoSave?.invalidateOnUnmount && autosaved) {
+        invalidate({
+          id,
+          invalidates: props.invalidates || ["list", "many", "detail"],
+          dataProviderName: props.dataProviderName,
+          resource: identifier,
+        });
+      }
+    };
+  }, [props.autoSave?.invalidateOnUnmount, autosaved]);
+
+  const submit = async (values: TVariables, isAutosave = false) => {
+    const isPessimistic = mutationMode === "pessimistic";
+
+    // Disable warning trigger when the form is being submitted
+    setWarnWhen(false);
+
+    // Redirect after a successful form submission
+    const onSuccessRedirect = (id?: BaseKey) => redirect(redirectAction, id);
+
+    const submissionPromise = new Promise<
+      // biome-ignore lint/suspicious/noConfusingVoidType: Void is an expected case for this promise.
+      CreateResponse<TResponse> | UpdateResponse<TResponse> | void
+    >((resolve, reject) => {
+      // Reject the mutation if the resource is not defined
+      if (!resource) return reject(missingResourceError);
+      // Reject the mutation if the `id` is not defined in edit action
+      if (isEdit && !id) return reject(missingIdError);
```

Comment:
I think we shouldn't reject here. AFAIk, it wasn't rejected before. And users may have a case where they update resource without an id e.g.: `api/users/settings`.

Reply:
```suggestion
      if (isClone && !id) return reject(missingIdError);
```

I think we should fail if clone is missing id tho.